netman/adapters/switches/juniper/mx.py

```diff
@@ -144,7 +147,66 @@ def edit_bond_spanning_tree(self, number, edge=None):
 
     def add_vrrp_group(self, vlan_number, group_id, ips=None, priority=None, hello_interval=None, dead_interval=None,
                        track_id=None, track_decrement=None):
-        raise NotImplementedError()
+        config = self.query(one_interface_vlan(vlan_number))
+
+        if len(config.xpath("data/configuration/interfaces/interface/unit")) < 1:
+            raise UnknownVlan(vlan_number)
+
+        for addr_node in config.xpath("data/configuration/interfaces/interface/unit/family/inet/address/name"):
+            address = IPNetwork(addr_node.text)
+            if all(ip in address for ip in ips):
```

Comment:
Would it be easier to read like this

    for ip in ips:
        if ip not in address:
            raise UnknownIP(",".join(map(str, ips)))

Reply:
Not